src/context/UserDataContext/properties/userProgressOnResources.ts

```diff
@@ -52,30 +56,26 @@ export default class UserProgressOnResourcesProperty extends UserDataPropertyAPI
     return {
       userProgressOnResources: this.progressValue,
       setUserProgressOnResources: (problemId, status) => {
-        if (!this.firebaseUserDoc) {
-          // if the user isn't using firebase, it is possible that they
-          // have multiple tabs open, which can result in localStorage
-          // being out of sync.
-          this.initializeFromLocalStorage();
-        }
+        problemId = replaceIllegalFirebaseCharacters(problemId);
+        // if the user isn't using firebase, it is possible that they
+        // have multiple tabs open, which can result in localStorage
+        // being out of sync.
+        // also bc of data loss let's just do this all the time
+        this.initializeFromLocalStorage();
         try {
           this.progressValue[problemId] = status;
 
           if (this.firebaseUserDoc) {
-            setDoc(
-              this.firebaseUserDoc,
-              {
-                [this.progressStorageKey]: {
-                  [problemId]: status,
-                },
-              },
-              { merge: true }
-            );
+            updateDoc(this.firebaseUserDoc, {
```

Comment:
what's the difference?

Reply:
Actually, I realized there is no difference 🤡 

I thought `setDoc` might not work with nested objects but actually that is incorrect (it does work with nested objects). But I think `arrayUnion` only works with `updateDoc` (but I could be wrong)